template.c

```diff
@@ -485,7 +485,7 @@ doEscape(uchar **pp, rs_size_t *pLen, unsigned short *pbMustBeFreed, int mode)
 		} else if((mode == SQL_ESCAPE) && *p == '\\') {
 			CHKiRet(cstrAppendChar(pStrB, '\\'));
 			iLen++;	/* reflect the extra character */
-		} else if((mode == JSON_ESCAPE) && (*p != '"' || *p != '\' )) {
+		} else if((mode == JSON_ESCAPE) && (*p != '"' || *p != '\\' )) {
```

Comment:
looks like I picked the wrong commit, this is also true for the later one. Sorry for that. 

BTW: you should squash those commits to make the PR easier to follow. If you don't know how, you can also mail me privately and I'll help :-)